File: java/sprykerFileCreator/model/definition/spryker/YvesControllerDefinition.java
package sprykerFileCreator.model.definition.spryker;

import pav.sprykerFileCreator.model.definition.ClassDefinitionInterface;
import pav.sprykerFileCreator.model.generator.SprykerConstants;

public class YvesControllerDefinition implements ClassDefinitionInterface {

    @Override
    public String getClassType() {
        return SprykerConstants.YVES_CONTROLLER;
    }

    @Override
    public String getNamePattern() {
        return "{{ className }}Controller";
    }

    @Override
    public String getNamespacePattern() {
        return "{{ projectName }}\\Yves\\{{ bundleName }}\\Controller";
    }

    @Override
    public String getMethodForReturnType() {
        return "";
    }

    @Override
    public String[] getDocBlockClasses() {
        String [] docBlockClasses = { SprykerConstants.YVES_FACTORY, SprykerConstants.CLIENT };

        return docBlockClasses;
    }

    @Override
    public String getDefaultParentClass() {
        return "Spryker\\Yves\\Application\\Controller\\AbstractController";
    }
}

File: src/main/kotlin/com/github/tobi812/sprykerplugin/services/definitions/spryker/YvesControllerDefinition.kt
package com.github.tobi812.sprykerplugin.services.definitions.spryker

import com.github.tobi812.sprykerplugin.constants.SprykerConstants
import com.github.tobi812.sprykerplugin.services.definitions.ClassDefinitionInterface

class YvesControllerDefinition : ClassDefinitionInterface {
    override val classType: String
        get() = SprykerConstants.YVES_CONTROLLER
    override val namePattern: String
        get() = "{{ className }}Controller"
    override val namespacePattern: String
        get() = "{{ projectName }}\\Yves\\{{ bundleName }}\\Controller"
    override val methodForReturnType: String
        get() = ""
    override val docBlockClasses: Array<String>
        get() = arrayOf(SprykerConstants.YVES_FACTORY, SprykerConstants.CLIENT)
    override val defaultParentClass: String
        get() = "Spryker\\Yves\\Application\\Controller\\AbstractController"
}
